### DISCUSSION TITLE
Where is my access log?

### DISCUSSION THREAD

**Discussion by saintcore** [nextcloud/all-in-one#1563] [Questions] [answered]
I recently moved from a classic nextcloud installation to aio and do only start to get comfortable with docker. So sry if this question is somehow "stupid". 

My basic setup is working including reverse-proxy setup, but as stupid as it sounds: I'm failing to find my access.log for the apache container. As far as I can see "docker logs nextcloud-aio-apache" should print them (or the corresponding docker json-log file which can be identified by 'docker inspect --format='{{.LogPath}}' nextcloud-aio-apache'). In my case, the mentioned logs do only show the errors, but no normal access.log entries are there.

Is there anything I'm missing here? I'm aware of how the apache2 logs are redirect ("The official httpd driver changes the httpd application’s configuration to write its normal output directly to /proc/self/fd/1 (which is STDOUT) and its errors to /proc/self/fd/2 (which is STDERR).") and checked the configs within the nextcloud-aio-apache container. Seems all to be ok. 

If you ask me why I'm sure there are only errors showing up with 'docker logs nextcloud-aio-apache' command: During my initial testing for reverse-proxy setup I've seen a few errors there because of a missing "trusted_proxies" entry within nextcloud config.php. As soon as I've added the trusted_proxies entry the errors were gone. So I was somehow able to see "error" logs for the not working connection, but can't see any logs for the now working connection.

**Comment by szaimen** [comment, marked as answer]
Hi, not sure why you need the access log?

As you can see here is the stdout for apache disabled so it will not be logged to the container logs. The reason for this is that it makes debugging much easier if only errors are logged. I hope this makes sense.

See
https://github.com/nextcloud/all-in-one/blob/70a03ba1ae2c0b161165ceaf89930a100f9aae15/Containers/apache/supervisord.conf#L12-L13

**Reply by saintcore** [reply]
Ok this does explain why I wasn't able to see the logs, thanks for clarification. I was confused by the following line:
https://github.com/nextcloud/all-in-one/blob/70a03ba1ae2c0b161165ceaf89930a100f9aae15/Containers/apache/nextcloud.conf#L6

However I would still like to have an access.log somehow. What would be the "docker-way" to get there? Just re-enabling the mentioned lines in supervisord.conf via (as example) mapping in a docker compose file?

As you asked why I need them: I like to use them for things like log analysing and so on. You can learn a lot via selfeducating when using tools like elk-stack, grafana and so on. Access-logs have been, at least for me, some great resource for those purposes.

Oh and not to forget: A huge :+1:  to you from me. You seem to be always respectful in your answers and try to help people to learn something and not just getting a solution :1st_place_medal:

**Comment by szaimen** [comment]
> However I would still like to have an access.log somehow. What would be the "docker-way" to get there? Just re-enabling the mentioned lines in supervisord.conf via (as example) mapping in a docker compose file?

There is currently no way to do so.

> As you asked why I need them: I like to use them for things like log analysing and so on. You can learn a lot via selfeducating when using tools like elk-stack, grafana and so on. Access-logs have been, at least for me, some great resource for those purposes.

I see. However since Apache is running behind a Caddy (that does the proxying and certificate stuff), you would not see much useful things in the logs anyway. My recommendation for you would be installing a Reverse Proxy and getting AIO to run behind that. Then you have all possibilities to define your own access logs and such. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md for how to install AIO behind a reverse proxy.

> Oh and not to forget: A huge 👍 to you from me. You seem to be always respectful in your answers and try to help people to learn something and not just getting a solution 🥇

Thanks a lot! :)

**Reply by saintcore** [reply]
I do already run my AIO-Instance behind a HAProxy instance as reverse proxy. However, as this is running in a shared environment, I need to ensure traffic between HAProxy- and NCAIO-Hosts is encrypted. Easiest way to achive this was to let your caddy-instance in place and ensure the same certificates (including private key) are in place at the haproxy-instance, decrypt the ssl-traffic (for header-modification and scanning --> https://doc.crowdsec.net/docs/next/bouncers/haproxy/ ) and encrypt outgoing traffic to ncaio with the same cert and private key again. This way I can ensure A) traffic between HAProxy and NCAIO is encrypted, B) I do not need to change anything manually at NCAIO, I just can run for example the docker run command in README.md and everything will work.

And lust but not least  you're of course right. I do have enhanced logs there (HAProxy) for each connection and I should be able to change the log format to be compatible with default apache log formats. So problem solved thanks :+1:
